build.sbt

```diff
@@ -22,19 +22,19 @@ ThisBuild / scalafixAll / skip := tlIsScala3.value
 ThisBuild / ScalafixConfig / skip := tlIsScala3.value
 ThisBuild / circeRootOfCodeCoverage := Some("rootJVM")
 
-val catsVersion = "2.10.0"
-val jawnVersion = "1.5.1"
-val shapelessVersion = "2.3.11"
+val catsVersion = "2.12.0"
+val jawnVersion = "1.6.0"
+val shapelessVersion = "2.3.12"
 val refinedVersion = "0.9.29"
 val refinedNativeVersion = "0.11.1"
```

Comment:
I don't know the historical reason for those two being different, but hopefully the build can be simplified now that **0.11.2** is published for all the base versions needed for this PR: https://github.com/fthomas/refined/pull/1274#issuecomment-2143510430

Reply:
the reason is binary compatibility.
We need to do #2276